README.md

```diff
@@ -79,6 +79,18 @@ The clusterconnected controller in Rancher uses the established tunnel to check
 that clusters are still responsive and sets alert conditions on the cluster
 object if they are not.
 
+#### HA operation (peering)
+
+remotedialer supports a mode where multiple servers can be configured as peers.
+In that mode all servers maintain a mapping of all client connections to all other
+servers, and can forward connections appropriately.
+
+Therefore, http requests to any of the clients can be resolved by any of the peer
+servers. This is useful for high availability, and Rancher leverages that
+functionality among replicas of the main pod.
```

Comment:
Here's my mental model based on code exploration and the information provided in Slack and in the PR.

![image](https://github.com/rancher/remotedialer/assets/7152146/afdae755-995b-4352-8b92-9f270fffd7fb)

Both the UI and the agents in downstream clusters will need to go through the load balancers in front of rancher. In case of an HA installation of rancher, we cannot be sure that the UI will be connected on the same "peer" as the downstream remotedialer client. For this reason, we must create a mesh of peers and proxy requests/response to the right peer.

nit: I found this paragraph a bit confusing. Which clients are we talking about here, the UI (well, for rancher anyway) or the "remotedialer" clients? I guess technically both, since the UI make request to some resources, but the remotedialer also make "requests" to peers in the cluster (eg: connect). 

What confused me as well was that not every requests can be completed by any peers in the sense that if Peer A is down, and you connect to Peer B, but the remotedialer client is connected to Peer A, then B cannot complete the request (Well.. I'm not too sure about how remotedialer handles the different failure scenarios so heh).

Reply:
All good, thanks! Approved